training/training_code/f1.py

```diff
@@ -0,0 +1,220 @@
+            precision and recall. Default value is 1.
+        threshold: Elements of `y_pred` greater than threshold are
+            converted to be 1, and the rest 0. If threshold is
+            None, the argmax is converted to 1, and the rest 0.
+    Returns:
+        F-Beta Score: float
+    Raises:
+        ValueError: If the `average` has values other than
+        [None, micro, macro, weighted].
+        ValueError: If the `beta` value is less than or equal
+        to 0.
+    `average` parameter behavior:
+        None: Scores for each class are returned
+        micro: True positivies, false positives and
+            false negatives are computed globally.
+        macro: True positivies, false positives and
+            false negatives are computed for each class
+            and their unweighted mean is returned.
+        weighted: Metrics are computed for each class
+            and returns the mean weighted by the
+            number of true instances in each class.
+    """
+
+    def __init__(self,
+                 num_classes,
+                 average=None,
+                 beta=1.0,
+                 threshold=None,
+                 name='fbeta_score',
+                 dtype=tf.float32):
+        super(FBetaScore, self).__init__(name=name)
+
+        if average not in (None, 'micro', 'macro', 'weighted'):
+            raise ValueError("Unknown average type. Acceptable values "
+                             "are: [None, micro, macro, weighted]")
+
+        if not isinstance(beta, float):
+            raise TypeError("The value of beta should be a python float")
+
+        if beta <= 0.0:
+            raise ValueError("beta value should be greater than zero")
+
+        if threshold is not None:
+            if not isinstance(threshold, float):
+                raise TypeError(
+                    "The value of threshold should be a python float")
+            if threshold > 1.0 or threshold <= 0.0:
+                raise ValueError("threshold should be between 0 and 1")
+
+        self.num_classes = num_classes
+        self.average = average
+        self.beta = beta
+        self.threshold = threshold
+        self.axis = None
+        self.init_shape = []
+
+        if self.average != 'micro':
+            self.axis = 0
+            self.init_shape = [self.num_classes]
+
+        def _zero_wt_init(name):
+            return self.add_weight(
+                name,
+                shape=self.init_shape,
+                initializer='zeros',
+                dtype=self.dtype)
+
+        self.true_positives = _zero_wt_init('true_positives')
+        self.false_positives = _zero_wt_init('false_positives')
+        self.false_negatives = _zero_wt_init('false_negatives')
+        self.weights_intermediate = _zero_wt_init('weights_intermediate')
+
+    # TODO: Add sample_weight support, currently it is
+    # ignored during calculations.
+    def update_state(self, y_true, y_pred, sample_weight=None):
```

Comment:
I am pretty sure that supporting the `sample_weight` mask is required for our case in order to get a proper result - otherwise it's computing the score including padding...

Reply:
we don't have `sample_weight` in addons. I wrote the first version of this metric and added it to addons. I did not include sample weights. I don't think we anyone took that up and updated the code.